### PR TITLE
LPS-160860 [Raylife AP] Create Dashboard Whats New Settings

### PR DESCRIPTION
**Pull request by joanagrego**
Related with --> [LPS-153065](https://issues.liferay.com/browse/LPS-153065)

### CODE REVIEW

**Comment by joanagrego** [comment]
ci:test:sf

**Comment by joanagrego** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+			active: false,
+			name: 'am',
+		},
+		{
+			active: false,
+			name: 'pm',
+		},
+	];
+
+	const [buttons, setButtons] = useState<cardButtonType[]>(
```
Comment (eliasantosf) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx:97]: Specify the name of the buttons.

**Review by eliasantosf — APPROVED** [review]
LGTM with some changes! @joanagrego

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+					{dataHoursOptions.map((item, index) => (
+						<ClaySelect.Option
+							key={index}
+							label={item.label}
+							value={item.value}
+						/>
+					))}
+				</ClaySelect>
+
+				{periodButtons.map((period, index) => {
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx:187]: Please change the parameter name here and put the parameter types

```suggestion
				{periodButtons.map((periodButton:cardButtonType, index:number) => {
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+			</div>
+
+			<hr className="mb-5"></hr>
+
+			{realUpdateToggled === false && <ScheduleUpdates />}
+
+			<div className="font-weight-bolder mt-3 pb-2 px-5">Display</div>
+
+			<div className="font-weight-lighter mt-3 pb-5 px-5 text-neutral-8">
+				<div className="d-flex dashboard-whats-new-modal-lines dashboard-whats-new-modal-tittle justify-content-between mb-3">
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx:229]: Please change the tittle name to title here and do the same everywhere it is used

```suggestion
				<div className="d-flex dashboard-whats-new-modal-lines dashboard-whats-new-modal-title justify-content-between mb-3">
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/index.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+			<Modal
+				Buttons={() => <ButtonsForSettingsModal />}
+				// eslint-disable-next-line react/no-children-prop
+				children={<WhatsNewStatus />}
+				modalStyle="modal-clay"
+				observer={observer}
+				size="md"
+				title="Settings"
+				visible={visible}
+			></Modal>
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/index.tsx:66]: Please, remove the children prop and do it as bellow

```suggestion
			<Modal
				Buttons={() => <ButtonsForSettingsModal />}
				modalStyle="modal-clay"
				observer={observer}
				size="md"
				title="Settings"
				visible={visible}
			>
			    <WhatsNewStatus />
			</Modal>
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-whats-new/index.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-undef */
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-whats-new/index.js:1]: If this is not necessary, please remove it.

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx (hunk 1 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/styles/apps/dashboard/whats-new-modal/index.scss [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+		width: 538px;
+	}
+
+	.dashboard-whats-new-modal-options {
+		min-width: 100px;
+	}
+
+	.dashboard-whats-new-modal-active-buttons {
+		background-color: rgba(76, 132, 255, 0.1);
+		border: 1px solid #4c84ff;
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/styles/apps/dashboard/whats-new-modal/index.scss:35]: We need to test it :)

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+	const selectPeriodCard = (currentIndex: number) => {
+		periodOfTheDay[currentIndex].active = !periodOfTheDay[currentIndex]
+			.active;
+
+		setPeriodButtons(periodOfTheDay);
+	};
+
+	const ScheduleUpdates = () => (
+		<div className="dashboard-whats-new-modal-container mt-3 pb-2 px-5">
+			<div className="font-weight-bolder"> Schedule</div>
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+
+			<div className="font-weight-semi-bold mt-4">Data from</div>
+
+			<div className="d-flex dashboard-whats-new-modal-buttons dashboard-whats-new-modal-lines mt-3">
+				<ClaySelect
+					aria-label="Select Label"
+					className="dashboard-whats-new-modal-period-options mr-3"
+					id="mySelectId"
+				>
+					{dataFromOptions.map((item, index) => (
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+
+	const [buttons, setButtons] = useState<cardButtonType[]>(
+		daysOfTheWeekButtons
+	);
+	const [periodButtons, setPeriodButtons] = useState<cardButtonType[]>(
+		periodOfTheDay
+	);
+
+	const selectCardDaysOfWeek = (currentIndex: number) => {
+		const newButtons = buttons.map((button, index) => {
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+	const ScheduleUpdates = () => (
+		<div className="dashboard-whats-new-modal-container mt-3 pb-2 px-5">
+			<div className="font-weight-bolder"> Schedule</div>
+
+			<div className="font-weight-semi-bold mb-3 mt-3">
+				Generate list on
+			</div>
+
+			<div className="d-flex dashboard-whats-new-modal-buttons flex-row">
+				{buttons.map((button, index) => (
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+						/>
+					))}
+				</ClaySelect>
+
+				<ClaySelect
+					aria-label="Select Label"
+					className="dashboard-whats-new-modal-options mr-3"
+					id="mySelectId"
+				>
+					{dataHoursOptions.map((item, index) => (
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx (hunk 2 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx (hunk 3 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/WhatsNewSettings.tsx]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/index.tsx (hunk 4 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/pages/SettingsModals/index.tsx]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-whats-new/index.js (hunk 5 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-whats-new/index.js]: Done.

**Comment by joanagrego** [comment]
ci:test:sf

**Review by silasbsi — APPROVED** [review]

**Comment by silasbsi** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward

**Comment by joanagrego** [comment]
ci:forward

**Comment by joanagrego** [comment]
ci:forward

**Comment by joanagrego** [comment]
ci:forward

**Comment by silasbsi** [comment]
ci:forward

**Pull request closed without merging**